File: http/src/main/java/webconnect/com/webconnect/observer/ErrorLiveData.java
package webconnect.com.webconnect.observer;


import android.arch.lifecycle.MutableLiveData;

public class ErrorLiveData extends MutableLiveData<String> {

    private static volatile ErrorLiveData sInstance;

    private ErrorLiveData() {

    }

    public static ErrorLiveData getInstance() {
        if (sInstance == null) {
            synchronized (ErrorLiveData.class) {
                if (sInstance == null) {
                    sInstance = new ErrorLiveData();
                }
            }
        }
        return sInstance;
    }
}

File: http/src/main/java/webconnect/com/webconnect/observer/ErrorLiveData.kt
package webconnect.com.webconnect.observer


import android.arch.lifecycle.LiveData
import android.arch.lifecycle.MutableLiveData

object ErrorLiveData {

    internal val error = MutableLiveData<String>()

    @JvmStatic
    fun getErrorLiveData(): LiveData<String> {
        return error
    }
}
